Remove extra semicolon that was being rendered on every route

kdaviduik opened the pull request:
### WHY are these changes introduced?
Remove extra semicolon that was being rendered on every route

This wasn't caught in CI since it's technically valid text, and was very easy to miss visually

<img width="642" height="533" alt="image" src="https://github.com/user-attachments/assets/db0d2883-8ee7-4161-89a4-8567b79115f6" />

I have another PR stacked on top of this one to add an eslint rule to prevent this type of issue from recurring without failing CI

#### Checklist

- [ ] I've read the [Contributing Guidelines](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md)
- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've added a [changeset](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md#changesets) if this PR contains user-facing or noteworthy changes
- [ ] I've added [tests](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md#testing) to cover my changes
- [ ] I've added or updated the documentation

<!--
 THANK YOU for your pull request! Members from the Hydrogen team will review these changes and provide feedback as soon as they are available.
-->

kdaviduik committed:
Remove extra semicolon that was being rendered on every route

This wasn't caught in CI since it's technically valid text, and was
very easy to miss visually

kdaviduik commented:
* **#3187** <a href="https://app.graphite.dev/github/pr/Shopify/hydrogen/3187?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#3186** <a href="https://app.graphite.dev/github/pr/Shopify/hydrogen/3186?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/Shopify/hydrogen/3186?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

juanpprieto commented:
Could we combine it with the https://github.com/Shopify/hydrogen/pull/3187 so we fix it properly via the new eslint rule? this way we could update recipes once

kdaviduik commented:
> Could we combine it with the https://github.com/Shopify/hydrogen/pull/3187 so we fix it properly via the new eslint rule? this way we could update recipes once

@juanpprieto yep, if you folks are happy with my solution in the other PR! I split them up just in case anyone disagreed with my approach for how we want to prevent these (with CI failures) in the future, so that this change could still ship easily

kdaviduik committed:
Add ESLint rule to prevent accidental single-character JSX text (#3187)

* Add ESLint rule to prevent accidental single-character JSX text

This custom rule catches stray semicolons and commas that appear as JSX text,
which are almost always unintentional. A trailing semicolon recently made it
to production in our skeleton template and was rendered on every route.

Since these are technically valid JSX text content, no existing ESLint rules
catch them. This custom rule specifically targets single-character text nodes
that are likely typos (currently ';' and ',').

The rule is added at the root project level rather than in individual
templates to ensure consistent coverage across all packages.

* Fix stray semicolons in test files caught by new ESLint rule

The new ESLint rule immediately caught two existing bugs where semicolons
were accidentally placed after JSX elements in test files. These semicolons
would have been rendered as text content in the DOM.

This validates the usefulness of the custom rule - it found real issues
that would have been visible in the rendered output during tests.

* Updating b2b recipe

* Updating express recipe

* Updating gtm recipe

* Updating markets recipe

* Updating partytown recipe

---------

Co-authored-by: rbshop <rheese.burgess@shopify.com>

kdaviduik commented:
# Note
I just merged in [this PR](https://github.com/Shopify/hydrogen/pull/3187)'s branch
